provider/search-provider/src/main/java/dev/ikm/tinkar/provider/search/Searcher.java

```diff
@@ -344,4 +345,68 @@ public static List<PublicId> getAllowedResultsFromResultConformance(NavigationCa
         return allowedResultsList;
     }
 
+    /**
+     * Returns a list of PublicIds for Descendants given an ancestor and list of ancestors to constrain by
+     *
+     * @param ancestorId            ID of the primary ancestor to find Descendants of
+     * @param constraintAncestorIds List of IDs for descendants to constrain by (includes itself)
+     * @return List of PublicIds for a given ancestor constrained by other ancestors
+     */
+    public static List<PublicId> getDescendantsWithConstraint(PublicId ancestorId, List<PublicId> constraintAncestorIds) {
+        List<PublicId> descendantsWithConstraints = new ArrayList<>();
+        if (PrimitiveData.get().hasPublicId(ancestorId)) {
+            EntityService.get().getEntity(ancestorId.asUuidArray()).ifPresent(ancestorEntity -> {
+                if (ancestorEntity instanceof Concept) {
+                    List<PublicId> constraintDescendants = new ArrayList<>();
+                    descendantsWithConstraints.addAll(descendantsOf(ancestorId));
+                    constraintAncestorIds.forEach(constraintId -> {
+                        if (PrimitiveData.get().hasPublicId(constraintId)) {
+                            EntityService.get().getEntity(constraintId.asUuidArray()).ifPresent(constraintEntity -> {
+                                if (constraintEntity instanceof Concept) {
```

Comment:
I know Ryan went over this with us, but I'm still fuzzy on how this all works. Is the idea that if an entity isn't a Concept it must be a constraint? So we remove all of the Concepts from the list of descendants and whatever is left are constraints?